### ISSUE TITLE
Fix FRM4SM_release_2 branch

### ISSUE THREAD

**Issue by pstradio** [awst-austria/qa4sm#535]
@sheenaze , @wpreimes  Unfortunately I pushed the new environment with Pytesmo==0.14.2 directly to the branch: https://github.com/awst-austria/qa4sm/tree/frm4sm-release-2.

Since the new error handling is included in the pytesmo release, a couple of tests have been broken. @s-scherrer needs to update QA4SM anyways, so they will be fixed soon.

**Comment by sheenaze** [comment]
Ok. Let me know if you need my help.

**Comment by pstradio** [comment]
@sheenaze @s-scherrer since there was recently an exchange of emails about this, I would like to point out that the tests were broken here. I don't think its a big deal, after samuel has worked on the qa4sm part the tests will be fixed and we can deploy. Monika, in your local environment you can just use pytesmo==0.14.1

**Comment by sheenaze** [comment]
It's not my local environment the problem here. I need to merge my changes to the frm4sm branch and push everything to the test instance, and I'm afraid that if I downgrade the pytesmo version, changes applied by you won't work :(

Generally it's not a big deal, I just got annoyed by all the errors I got on Thursday and at some point I couldn't differentiate which came from my changes and which from the pytesmo change, but it looks much better now.
Generally if you take a look at tests, there are still the same two failing: https://github.com/sheenaze/qa4sm/actions/runs/3758002622/jobs/6385828334
but the reason for failing changed, when I set 'ignore'. 

I'll be back to work on Wednesday and I'll push everything to the test instace anyway, to test changes I applied, and whenever you fix the problems just push them to the frm4sm branch and deploy or let me know and I'll do deployment. 

And now just enjoy the holidays :)

**Comment by s-scherrer** [comment]
I just ran the `test_validation_tcol` test on pytesmo 0.14.1 (old) and 0.14.2 (new).

Both essentially show the same behaviour, namely that the validation works on 4 out of the 9 requested points, and fails on the others because one of the datasets doesn't have data.

In the old pytesmo version, the remaining 5 gpis are silently dropped, and the results file will only contain results for the 4 valid gpis (even though run.ok_points is set to 9, because pytesmo didn't throw an error). In this case the NaN ratio is 0 (none of the results is NaN, because the skipped gpis are not taken into account).

The new pytesmo version now captures these errors and returns a dummy result for them (basically NaN everywhere except for the metadata like gpi, lat, lon, status). The new results file therefore has 9 entries, one for each of the validation gpis, but has NaNs for 5 out of 9 gpis, which makes the NaN-ratio test fail.

So in principle the new error handling works as it should, but I still need to adapt a few things (e.g. a proper assignment of ok_points/error_points). However, even with these things implemented the tests will fail. We need to either change the allowed NaN ratio (https://github.com/awst-austria/qa4sm/blob/aa1507122ca93d8e28862d78adc2aac368a80ad9/validator/tests/test_validation.py#L161), or use a different test setup.

**Comment by s-scherrer** [comment]
I accidentally pushed my changes directly to the frm4sm-release-2 branch, sorry for that. But it looks like the 2 failing tests are running now. The exact changes can be found here: https://github.com/awst-austria/qa4sm/commit/1583bb729a240fb7fb2859dec83247df2aed5132

The main change is that we now set "run.ok_points" based on the returned error codes. If any of the comparisons for a gpi fails, the point is counted as error_point (but the results are still accessible if e.g. triple collocation fails but the pairwise comparisons work). Does that sound reasonable to you?

As mentioned above, I had to adapt the NaN-ratio to allow up to 0.75. @pstradio I also saw that in the test_c3s_validation_upscaling only 4 out of 16 points actually had data for C3S (spatial ref) and upscaled ISMN, is this expected? This was already the case with the old pytesmo version, but with the old error handling the missing points were dropped and ok_points was nevertheless set to 16.

**Comment by s-scherrer** [comment]
I just saw that the long-running tests are failing due to differences in the error_points, which is somewhat expected, but needs some investigation. I'll try to fix them and make a new PR then.

**Comment by sheenaze** [comment]
I didn't have time today to update tests to verifying changes I made. I'm planning on finishing it tomorrow and I'll take a look at those failing tests as well.

**Comment by s-scherrer** [comment]
Once #536 is merged these tests should be fixed.

**Comment by sheenaze** [comment]
Yeah, I noticed that, but I want to take a look at them anyway, because I kind of should know what happens there :)

**Comment by sheenaze** [comment]
I want to merge #536, but it would be good if the question about the numbers of valid points in the test test_validation_smap_ref is still addressed. So, @pstradio or @wpreimes, if you can answer why the number of valid points changed and if it's expected or not, please answer here.

**Comment by sheenaze** [comment]
@pstradio  Can we close this issue?
